qiskit_nature/second_q/mappers/bosonic_mapper.py

```diff
@@ -0,0 +1,62 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Bosonic Mapper."""
+
+from __future__ import annotations
+
+from qiskit.quantum_info import SparsePauliOp
+
+from qiskit_nature.second_q.operators import BosonicOp
+
+from .qubit_mapper import ListOrDictType, QubitMapper
+
+
+class BosonicMapper(QubitMapper):
+    """Mapper of Bosonic Operator to Qubit Operator
+
+    The following attributes can be read and updated once the ``BosonicMapper`` object
+    has been constructed.
+
+    """
+
+    def __init__(self, max_occupation: int) -> None:
+        """
+        Args:
+            max_occupation: defines the excitation space of the k-th bosonic state. Together with the
+                number of modes required to represent the bosonic operator, it defines the minimum length
+                of the qubit register. The minimum value is 1.
+        """
+        super().__init__()
+        self.max_occupation = max_occupation
+
+    @property
+    def max_occupation(self) -> int:
+        """The max. occupation for the bosonic linear mapper"""
+        return self._max_occupation
+
+    @max_occupation.setter
+    def max_occupation(self, max_occupation: int) -> None:
+        if max_occupation < 1:
+            raise ValueError(
+                "Max occupation for bosonic linear mapper must be at least 1. "
```

Comment:
```suggestion
                "The maximum occupation must be at least 1. "
```
Just because this generalizes to any bosonic mapper and not just the linear one